File: python/easy/9. Palindrome Number.py
# 9. Palindrome Number

def isPalindrome(x: int) -> bool:
  if(x < 0):
    return False # as negetive numbers will not be palindrome 

  temp = x
  check = 0
  while temp > 0:
    check *= 10
    check += temp % 10
    temp //= 10
  
  if check == x: 
     return True   

  return False

print(isPalindrome(1221))

File: java/easy/9. Palindrome Number.java
// 9. Palindrome Number

class Solution {
  public boolean isPalindrome(int x) {
      if (x < 0) return false;

      int a = x;
      int b = 0;
      
      while(a>0){
          b *= 10;
          b += a % 10;
          a /= 10;
      }

      if (b == x) return true;

      return false;
  }
}